### DISCUSSION TITLE
How to delete a row in exceljs?

### DISCUSSION THREAD

**Discussion by Yavdhesh** [exceljs/exceljs#1675] [Q&A] [answered]
I am stuck as how to delete a row in exceljs?

**Comment by briangonzalez** [comment, marked as answer]
If your row has an index of 30, you could simply run:

```
myWorksheet.spliceRows(30, 1)
```

Which reads "delete one row starting at index 30"

**Reply by Yavdhesh** [reply]
works fine, thank you :)

**Comment by hansithad** [comment]
I have a sheet with 21 rows, I want to delete all rows except header rows ( first row)
I was excepting this to work , but it's not. it has all the rows

`sheet.spliceRows(2, 20);`


if I try something like `sheet.spliceRows(2, 5);` it will remove some rows

any idea why ?

**Comment by pilaoda** [comment]
There are some bugs in `sheet.spliceRows` definitly.
